### PR TITLE
moving globalApiPrefix to Client

### LINKED ISSUE #80: Can't Use Multiple Versioned API Clients
I'm about to do this refactor and PR it but wanted to document why you would want to do this. There is a limitation of using the package level variable `globalApiPathPrefix`. If you call `goshopify.NewClient` using `WithVersion` it will always set all previous Clients to the last version passed. The global string should be refactored to a Client.pathPrefix and a first pass refactor could be

```
path := fmt.Sprintf("%s/%s.json", globalApiPathPrefix, locationsBasePath)
// becomes
path := fmt.Sprintf("%s/%s.json", s.client.pathPrefix, locationsBasePath)
```
but I would prefer the prefix was prepended in `Client.CreateAndDo` so you don't have to worry about it at the service level since the client should really dictate where all calls are going and you shouldn't be able to re-path which api version you're using in the Services but could be convinced otherwise. This is the version I will PR so if @gordcurrie or somebody else wants to stop me before I get there then please respond.

e.g. of how to recreate this
```
func newStableShopifyClient() *goshopify.Client {
	return goshopify.NewClient(shopifyApp, shopifyName, "")
}

func newVersionedShopifyClient(version string) *goshopify.Client {
	return goshopify.NewClient(shopifyApp, shopifyName, "", goshopify.WithVersion(version))
}

_ = newStableShopifyClient()
fmt.Println(goshopify.MetafieldPathPrefix("test", 1234))
_ = newVersionedShopifyClient("2019-04")
fmt.Println(goshopify.MetafieldPathPrefix("test", 1234))
```

returns something like this
```
admin/test/1234/metafields
admin/api/2019-04/test/1234/metafields
```

### PR DESCRIPTION
**Pull request by luthermonson**
New attempt with all latest changes from gomod included

Closes #80

### CODE REVIEW

**Comment by rickywiens** [comment]
Hey @luthermonson,

Just wondering if we can cover some of the new cases you added in the goshopify.go file?

https://codecov.io/gh/bold-commerce/go-shopify/pull/83/src/goshopify.go

I'm a bit less concerned about the oauth.go changes, like you mentioned the newClient change kinda messed that up and we can clean up those tests ourselves later.

**Comment by luthermonson** [comment]
@rickywiens ya im working on it just havent found time yet! didn't think this was gonna get merged till i fixed those but had other stuff come up. adding Client as a field on App was the hack to get around it but i think that GetAccessToken method should be on Client personally.

**Comment by luthermonson** [comment]
@rickywiens is this good enough or more done with oauth?

**Comment by rickywiens** [comment]
> @rickywiens is this good enough or more done with oauth?

I think this should be good - we can patch up the tests later.

It was brought up internally that we might want to alter the import in the gomod file:

https://github.com/golang/go/wiki/Modules#semantic-import-versioning

What do you think about that?

File: oauth.go [hunk 1]
```diff
@@ -46,8 +46,15 @@ func (app App) GetAccessToken(shopName string, code string) (string, error) {
...
 	}
 
-	client := NewClient(app, shopName, "")
+	client := app.Client
+	if client == nil {
+		client = NewClient(app, shopName, "")
+	}
+
 	req, err := client.NewRequest("POST", "admin/oauth/access_token", data, nil)
+	if err != nil {
```
Comment (gordcurrie) [inline oauth.go]: Thanks for adding this.

File: goshopify.go [hunk 2]
```diff
@@ -221,11 +234,22 @@ func (a App) NewClient(shopName, token string, opts ...Option) *Client {
 // token. The shopName parameter is the shop's myshopify domain,
 // e.g. "theshop.myshopify.com", or simply "theshop"
 func NewClient(app App, shopName, token string, opts ...Option) *Client {
-	httpClient := http.DefaultClient
+	baseURL, err := url.Parse(ShopBaseUrl(shopName))
+	if err != nil {
+		panic(err) // something really wrong with shopName
```
Comment (gordcurrie) [inline goshopify.go]: I'm torn about a panic here but totally agree it's better to check it than just ignore it.

Comment (luthermonson) [reply goshopify.go]: you reeeeeeally need to screw up to get this... i dont disagree panic is lame but adding an error return seemed like a pain in the ass, if you want the panic gone i'd suggest we go back to ignoring the error

**Comment by luthermonson** [comment]
> > @rickywiens is this good enough or more done with oauth?
> 
> I think this should be good - we can patch up the tests later.
> 
> It was brought up internally that we might want to alter the import in the gomod file:
> 
> https://github.com/golang/go/wiki/Modules#semantic-import-versioning
> 
> What do you think about that?

I think you mean changing this to `module github.com/bold-commerce/go-shopify/v3` at the top of go.mod? I'm against this, but it's your project. Adding the go.mod file didn't change the API of the package, just let the gomod nerds use it properly.

**Comment by luthermonson** [comment]
@rickywiens needed anything else for this?

**Comment by rickywiens** [comment]
Hey @luthermonson. I was just trying to figure out how to get around the codecov requirement. I'm merging it locally now and I will push it up.

**Pull request merged**